discovery_engine_core/core/src/stack/filters/article.rs

```diff
@@ -39,52 +39,70 @@ impl ArticleFilter for DuplicateFilter {
     fn apply(
         history: &[HistoricDocument],
         stack: &[Document],
-        articles: Vec<Article>,
+        mut articles: Vec<Article>,
     ) -> Result<Vec<Article>, GenericError> {
-        let urls = history
-            .iter()
-            .map(|doc| doc.url.as_str())
-            .chain(stack.iter().map(|doc| doc.resource.url.as_str()))
-            .collect::<HashSet<_>>();
+        // discard dups in the title keeping only the best ranked
+        articles.sort_unstable_by(|art1, art2| {
+            match art1.title.to_lowercase().cmp(&art2.title.to_lowercase()) {
```

Comment:
Maybe we can a function to normalize the article for comparison? here are not trimming while below we are.

Reply:
good catch. added a helper `normalize` to `utils`

Reply:
[now in `article`](https://github.com/xaynetwork/xayn_discovery_engine/pull/266#discussion_r835468158)